ProcessMaker/Traits/HideSystemResources.php

```diff
@@ -59,37 +59,41 @@ public function scopeNonSystem($query)
         if (substr(static::class, -8) === 'Category') {
             return $query->where('is_system', false);
         } elseif (static::class === Process::class) {
-            $systemCategory = ProcessCategory::where('is_system', true)->pluck('id');
-
-            return $query->whereNotIn('process_category_id', $systemCategory)
+            return $query
                 ->where('is_template', false)
                 ->when(Schema::hasColumn('processes', 'asset_type'), function ($query) {
                     return $query->whereNull('asset_type');
+                })
+                ->whereDoesntHave('categories', function ($query) {
+                    $query->where('is_system', true);
                 });
         } elseif (static::class === Screen::class) {
-            $systemCategory = ScreenCategory::where('is_system', true)->pluck('id');
-
-            return $query->whereNotIn('screen_category_id', $systemCategory)
-                    ->where('is_template', false)
-                    ->when(Schema::hasColumn('screens', 'asset_type'), function ($query) {
-                        return $query->whereNull('asset_type');
-                    });
+            return $query
+                ->where('is_template', false)
+                ->when(Schema::hasColumn('screens', 'asset_type'), function ($query) {
+                    return $query->whereNull('asset_type');
+                })
+                ->whereDoesntHave('categories', function ($query) {
```

Comment:
hi @agustinbusso this code displays the system screens (which should be hidden) can you try this query instead?

```
->whereHas('categories', function ($query) {
    $query->where('is_system', false);
})
```

![image](https://github.com/ProcessMaker/processmaker/assets/90741874/cd402279-7ba2-477c-ace1-d54863183edf)

Reply:
Good catch, thank you so much Eleazar!